
<launch>

    <!--descripcion del robot TF-->
    <param name="robot_description" command="$(find xacro)/xacro '$(find champy_2-0)/urdf/champy_2-0.urdf.xacro'"/>
    <node name="champy_robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" output="screen" required="true"/>
    
    <!-- RPLIDAR -->
    <node name="rplidarNode"          pkg="rplidar_ros"  type="rplidarNode" output="screen">
        <param name="serial_port"         type="string" value="/dev/ttyUSB0"/>  
        <param name="serial_baudrate"     type="int"    value="256000"/>
        <param name="frame_id"            type="string" value="Rplidar_link"/>
        <param name="inverted"            type="bool"   value="true"/>
        <param name="angle_compensate"    type="bool"   value="true"/>
    </node>

    <!-- CAMARA ZED-->

    <node pkg="robot_localization" type="ekf_localization_node" name="ekf_localization_map" clear_params="true">
        <rosparam command="load" file="$(find champy_2-0)/params/ekf/fusion.yaml" />
    </node>  
    <node name="zed" pkg="zed_wrapper" type="zed_wrapper_node" output="screen" required="true">
        <rosparam file="$(find champy_2-0)/params/camara_zed_2/zed.yaml" command="load" />
        <param name="svo_file"                          value="" />
        <param name="stream"                            value="" />
    </node>


     <!--ar_track_alvar -->
    <node name="ar_track_alvar" pkg="ar_track_alvar" type="individualMarkersNoKinect" respawn="false" output="screen">
        <param name="marker_size"           type="double" value="18" />
        <param name="max_new_marker_error"  type="double" value="0.14" /> <!-- 0.14 -->
        <param name="max_track_error"       type="double" value="0.16" /> <!-- 0.16 -->
        <param name="output_frame"          type="string" value="base_link" />
        <remap from="camera_image"  to="/zed/rgb_raw/image_raw_color" />
        <remap from="camera_info"   to="/zed/rgb_raw/camera_info" />
    </node>
    

    <node pkg="champy_2-0" type="wheel_odom.py" name="odom_wheel_node"/>
    
    <!--due_core -->
    <node pkg="rosserial_python" type="serial_node.py" name="serial_node">
        <param name="port" value="/dev/ttyACM0"/>
        <param name="baud" value="250000"/>
    </node>

    <!--<node pkg="champy_2-0" type="set_params.py" name="param_setter"
        <node pkg="champy_2-0" type="sub_app.py" name="sub_app"/>
        <node pkg="laser_filters" type="scan_to_scan_filter_chain" output="screen" name="laser_filter">
            <rosparam command="load" file="$(find champy_2-0)/params/lidar/lidar.yaml" />
        </node>
            <node name="rviz" pkg="rviz" type="rviz" args="-d $(find champy_2-0)/champy.rviz" output="screen" /> 

    />->-->
    <node pkg="champy_2-0" type="map.py" name="map_edit"/>
    <node pkg="champy_2-0" type="launch_realsense.py" name="launch_realsense_node"/>
    <node pkg="champy_2-0" type="launch_auto.py" name="launch_auto"/>
    <node pkg="champy_2-0" type="launch_mapeo.py" name="launch_mapeo"/>
    <node pkg="champy_2-0" type="enviodato_app.py" name="envio_app"/>
    <node pkg="rosserial_server" type="socket_node" name="tcp"/>

    <node name="laserscan_multi_merger" pkg="ira_laser_tools" type="laserscan_multi_merger" respawn="false" output="screen">
        <param name="destination_frame" value="base_link" />
        <param name="cloud_destination_topic" value="/cloud_fus" /> 
        <param name="scan_destination_topic" value="/scan_fus" /> 
        <param name="laserscan_topics" value="/scan_mod /scan_realsense" /> 
        <param name="angle_min" value="-3.1416" /> 
        <param name="angle_max" value="3.1416" /> 
        <param name="angle_increment" value="0.00871450919658" /> 
        <param name="scan_time" value="0.0995560958982" /> 
        <param name="range_min" value="0.03" /> 
        <param name="range_max" value="40.0" /> 
    </node>
    
    <node pkg="champy_2-0" type="copiaodo.py" name="relasense_lidar"/>

</launch>

